gax/retry_loop_test.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 Google Inc.  All rights reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gax/retry_loop.h"
+#include "google/longrunning/operations.pb.h"
+#include "googletest/include/gtest/gtest.h"
+#include "gax/backoff_policy.h"
+#include "gax/call_context.h"
+#include "gax/retry_policy.h"
+#include <chrono>
+
+namespace {
+using namespace ::google;
+
+std::unique_ptr<gax::RetryPolicy> ErrCountRetryFactory(int n) {
+  return std::unique_ptr<gax::LimitedErrorCountRetryPolicy>(
+      new gax::LimitedErrorCountRetryPolicy(n));
+}
+
+TEST(RetryLoop, Basic) {
+  // Hack to count the number of backoff attempts.
+  static int delay_count = 0;
```

Comment:
Please strongly consider getting rid from the static context here.

For example it can be done either by replacing `unique_ptr` with regular pointers for `MakeRetryCall` or by something like:
1) Declare `DummyBackofPolicy(int* delay_count)` constructor (pointer to indicate that the value is going to be modified from insid, referrence is not that "descriptive")
2) Redeclare `MakeDummyBackoff()` as `MakeDummyBackoff(int* delay_count)` and bypass it to the constructor
3) Use the pointer (class member) instead of the static value in `OnCompletion()`
4) Declare this class outside of the single test scope, so it can be reused by other tests (there will be more, I'm sure, retries is a complicated thing).

Reply:
I removed the static counter as we discussed offline.